apollo-at-link/src/location.rs

```diff
@@ -0,0 +1,1074 @@
+            None => panic!("Schema has no type \"{}\"", self.type_name),
+        }
+    }
+
+    pub fn try_get<'schema>(&self, schema: &'schema Schema) -> Option<&'schema Node<ScalarType>> {
+        match schema.types.get(&self.type_name) {
+            Some(ExtendedType::Scalar(ref type_)) => Some(type_),
+            _ => None,
+        }
+    }
+
+    pub fn make_mut<'schema>(&self, schema: &'schema mut Schema) -> &'schema mut Node<ScalarType> {
+        match schema.types.get_mut(&self.type_name) {
+            Some(ExtendedType::Scalar(ref mut type_)) => type_,
+            Some(_) => panic!("Schema type \"{}\" was not a scalar", self.type_name),
+            None => panic!("Schema has no type \"{}\"", self.type_name),
+        }
+    }
+
+    pub fn try_make_mut<'schema>(
+        &self,
+        schema: &'schema mut Schema,
+    ) -> Option<&'schema mut Node<ScalarType>> {
+        if self.try_get(schema).is_some() {
+            Some(self.make_mut(schema))
+        } else {
+            None
+        }
+    }
+}
+
+impl Display for ScalarTypeDefinitionLocation {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.type_name)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ObjectTypeDefinitionLocation {
+    pub type_name: Name,
+}
+
+impl ObjectTypeDefinitionLocation {
+    pub fn get<'schema>(&self, schema: &'schema Schema) -> &'schema Node<ObjectType> {
+        match schema.types.get(&self.type_name) {
+            Some(ExtendedType::Object(ref type_)) => type_,
+            Some(_) => panic!("Schema type \"{}\" was not an object", self.type_name),
+            None => panic!("Schema has no type \"{}\"", self.type_name),
+        }
+    }
+
+    pub fn try_get<'schema>(&self, schema: &'schema Schema) -> Option<&'schema Node<ObjectType>> {
+        match schema.types.get(&self.type_name) {
+            Some(ExtendedType::Object(ref type_)) => Some(type_),
+            _ => None,
+        }
+    }
+
+    pub fn make_mut<'schema>(&self, schema: &'schema mut Schema) -> &'schema mut Node<ObjectType> {
+        match schema.types.get_mut(&self.type_name) {
+            Some(ExtendedType::Object(ref mut type_)) => type_,
+            Some(_) => panic!("Schema type \"{}\" was not an object", self.type_name),
+            None => panic!("Schema has no type \"{}\"", self.type_name),
+        }
+    }
+
+    pub fn try_make_mut<'schema>(
+        &self,
+        schema: &'schema mut Schema,
+    ) -> Option<&'schema mut Node<ObjectType>> {
+        if self.try_get(schema).is_some() {
+            Some(self.make_mut(schema))
+        } else {
+            None
+        }
+    }
+}
+
+impl Display for ObjectTypeDefinitionLocation {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.type_name)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ObjectFieldDefinitionLocation {
```

Comment:
A lot of these seem to be wrappers around what's already in the compiler. What information is missing in the compiler that would make it easier to not have these wrappers/